registry/storage/blobwriter_resumable.go

```diff
@@ -2,8 +2,198 @@
 
 package storage
 
-import "github.com/docker/distribution/digest"
+import (
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"strconv"
 
-func (bw *blobWriter) setupResumableDigester() {
-	bw.resumableDigester = digest.NewCanonicalResumableDigester()
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/distribution/context"
+	storagedriver "github.com/docker/distribution/registry/storage/driver"
+	"github.com/stevvooe/resumable"
+
+	// register resumable hashes with import
+	_ "github.com/stevvooe/resumable/sha256"
```

Comment:
Presumably, the init function of this package should run after `crypto/sha256`. We can probably force this by making resumable depend on crypto.